apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/queries.ex

```diff
@@ -508,36 +497,20 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Queries do
     DatabaseQuery.call!(db_client, refresh_connected_query)
   end
 
-  defp get_connected_remaining_ttl(db_client, device_id) do
-    fetch_connected_ttl_statement = """
-    SELECT TTL(connected)
-    FROM devices
-    WHERE device_id=:device_id
-    """
+  defp get_connected_remaining_ttl(realm, device_id) do
+    keyspace_name =
+      CQLUtils.realm_name_to_keyspace_name(realm, Config.astarte_instance_id!())
 
-    fetch_connected_ttl_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(fetch_connected_ttl_statement)
-      |> DatabaseQuery.put(:device_id, device_id)
-      |> DatabaseQuery.consistency(:quorum)
-
-    with {:ok, result} <- DatabaseQuery.call(db_client, fetch_connected_ttl_query),
-         ["ttl(connected)": ttl] when is_integer(ttl) <- DatabaseResult.head(result) do
-      {:ok, ttl}
-    else
-      :empty_dataset ->
-        {:error, :device_not_found}
-
-      ["ttl(connected)": nil] ->
-        {:ok, 0}
-
-      %{acc: _, msg: error_message} ->
-        Logger.warning("Database error: #{error_message}.")
-        {:error, :database_error}
-
-      {:error, reason} ->
-        Logger.warning("Database error while retrieving property: #{inspect(reason)}.")
-        {:error, :database_error}
+    query =
+      Device
+      |> where(device_id: ^device_id)
+      |> select([device], fragment("TTL(?)", device.connected))
+      |> put_query_prefix(keyspace_name)
+
+    case Repo.fetch_one(query) do
+      n when is_number(n) -> {:ok, n}
+      nil -> {:error, :device_not_found}
+      {:error, reason} -> {:error, reason}
     end
```

Comment:
See comment in `repo.ex` for this, but this should be directly handled by `fetch_one`